src/libraries/System.Private.Xml/src/System/Xml/Cache/XPathNodeInfoAtom.cs

```diff
@@ -105,8 +106,8 @@ public XPathNodeInfoAtom(XPathNodePageInfo pageInfo)
         /// <summary>
         /// Construct a new shared information atom.  This method should only be used by the XNodeInfoTable.
         /// </summary>
-        public XPathNodeInfoAtom(string localName, string namespaceUri, string prefix, string baseUri,
-                                         XPathNode[] pageParent, XPathNode[] pageSibling, XPathNode[] pageSimilar,
+        public XPathNodeInfoAtom(string localName, string namespaceUri, string prefix, string? baseUri,
```

Comment:
In the constructor above should pageInfo be nullable?

Reply:
also since that constructor doesn't call Init and is public should all the fields marked with `null!` be actually nullable? (or check if the constructor is ever called and perhaps fix that callsite)

Reply:
> In the constructor above should pageInfo be nullable?

No, we never pass a nullable value, hence is fine being non-nullable.

> also since that constructor doesn't call Init and is public should all the fields marked with null! be actually nullable? 

As per the summary on line 98, they are only null for element in position 0 of `XPathNode[]` fields, I can make them nullable here but then I will have to add a lot of assertions and/or `!` since many places in the code assume that those fields are not null.